panther_hardware_interfaces/include/panther_hardware_interfaces/panther_system/robot_driver/roboteq_robot_driver.hpp

```diff
@@ -0,0 +1,180 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_ROBOT_DRIVER_ROBOTEQ_ROBOT_DRIVER_HPP_
+#define PANTHER_HARDWARE_INTERFACES_PANTHER_SYSTEM_ROBOT_DRIVER_ROBOTEQ_ROBOT_DRIVER_HPP_
+
+#include <chrono>
+#include <string>
+#include <vector>
+
+#include "panther_hardware_interfaces/panther_system/robot_driver/canopen_manager.hpp"
+#include "panther_hardware_interfaces/panther_system/robot_driver/robot_driver.hpp"
+#include "panther_hardware_interfaces/panther_system/robot_driver/roboteq_data_converters.hpp"
+#include "panther_hardware_interfaces/panther_system/robot_driver/roboteq_driver.hpp"
+
+namespace panther_hardware_interfaces
+{
+
+struct MotorNames
+{
+  static constexpr char LEFT[] = "left";
+  static constexpr char RIGHT[] = "right";
+};
+
+struct DriverNames
+{
+  static constexpr char DEFAULT[] = "default";
+  static constexpr char FRONT[] = "front";
+  static constexpr char REAR[] = "rear";
+};
+
+struct MotorChannels
+{
+  static constexpr std::uint8_t LEFT = RoboteqDriver::kChannel2;
+  static constexpr std::uint8_t RIGHT = RoboteqDriver::kChannel1;
+};
+
+/**
+ * @brief Abstract class implementing RobotDriver for robots using Roboteq drivers.
+ * It uses canopen_manager for communication with Roboteq controllers.
+ * This class implements the activation procedure for controllers, which involves resetting the
+ * script and sending an initial 0 command. It also provides methods to get data feedback and send
+ * commands. Data is converted between raw Roboteq formats and SI units using
+ * roboteq_data_converters.
+ */
+class RoboteqRobotDriver : public RobotDriver
```

Comment:
In case of making `RobotDriver` as a template class it can be:
```suggestion
class RoboteqRobotDriver : public RobotDriver<RoboteqData>
```